### DISCUSSION TITLE
Running the Messenger example of OpenDDS-3.21

### DISCUSSION THREAD

**Discussion by h3gui** [OpenDDS/OpenDDS#3669] [Q&A] [answered]
Hello,
After building the code with VS2019, a DCPSInfoRepo executable was created under C:\Users\Test\OpenDDS-3.21\bin rather than C:\Users\Test\OpenDDS-3.21\ACE_wrappers\bin.
Running %ACE_ROOT%\bin\DCPSInfoRepo results in file not found as can be seen in the attached files.
Also, simply copying and pasting the DCPSInfoRepo file into  C:\Users\Test\OpenDDS-3.21\ACE_wrappers\bin did not solve the problem.
Your help is appreciated.
Thanks
![command_prompt](https://user-images.githubusercontent.com/110626246/183215722-cd60bbd9-7b40-4de8-9b55-6e3ad33a625b.png)
![powershell](https://user-images.githubusercontent.com/110626246/183215738-8393f786-7b85-4131-a186-03de1a065d41.png)
![VS2019_powershell](https://user-images.githubusercontent.com/110626246/183215756-a19ff4b6-6fda-4aef-8918-2ef9edb9f648.png)

**Comment by mitza-oci** [comment, marked as answer]
Please post text instead of images.

Why are the `%` characters part of the file/directory names in the commands shown here?

**Reply by h3gui** [reply]
just following the quick start manual. $ sign is for Linux and % for windows

**Reply by mitza-oci** [reply]
Please see https://opendds.org/quickstart/GettingStartedWindows.html

If you want to start the DCPSInfoRepo process directly (without any other script), you can run `start DCPSInfoRepo` from the Visual Studio Command Prompt that has the OpenDDS development environment in effect (which can be the result of `configure` or `setenv`).

`%VAR%` is a command prompt string substitution mechanism.  If an environment variable named `VAR` is defined, the text `%VAR%` in a command string is expanded to the value of the environment variable.  It is similar to using `$VAR` in a Linux shell.

**Reply by h3gui** [reply]
Thank you Adam. start DCPSInfoRepo worked and created the repo.ior file in the same folder as DCPSInfoRepo which is C:\Users\Test\OpenDDS-3.21\ACE_wrappers\bin.  
By the way, DCPSInfoRepo was initially generated in C:\Users\Test\OpenDDS-3.21\bin and I simply copied it to C:\Users\Test\OpenDDS-3.21\ACE_wrappers\bin assuming that it should be there. 

However, the publisher and subscriber are in another folder C:\Users\Test\OpenDDS-3.21\DevGuideExamples\DCPS\Messenger and caused errors when running them from that folder or after copying them to C:\Users\Test\OpenDDS-3.21\ACE_wrappers\bin. 

C:\Users\Test\OpenDDS-3.21\ACE_wrappers\bin>start DCPSInfoRepo (created the rep.ior folder and opened a window)

C:\Users\Test\OpenDDS-3.21\ACE_wrappers\bin>start publisher (opened a window and showed this warning
WARNING: get_fully_qualified_hostname: Could not find FQDN. Using "W0052" as fully qualified hostname, please correct system configuration.)

C:\Users\Test\OpenDDS-3.21\ACE_wrappers\bin>start subscriber (opened a window, displayed something then the subscriber and publisher windows disappeared)

C:\Users\Test\OpenDDS-3.21\ACE_wrappers\bin>subscriber
(5880|8992) WARNING: get_fully_qualified_hostname: Could not find FQDN. Using "W0052" as fully qualified hostname, please correct system configuration. (after some time, abort has been called )

Your help is appreciated

**Reply by mitza-oci** [reply]
I'm not sure why using ACE's bin directory is part of this.  Let that be for ACE and put OpenDDS files elsewhere.

I'm not sure what errors you're referring to.  Warnings don't prevent the system from working (typically).  From what I've seen so far it probably worked.

Have you run run_test.pl script as shown on https://opendds.org/quickstart/GettingStartedWindows.html? After that is working you can start the programs without the help of the script.  The script prints the full command lines used to launch the test applications.  Also consider redirecting the stdout/stderr output to a file so it won't matter when each window closes, or keep the window open after the process exits.

**Reply by h3gui** [reply]
Thanks Adam. I had it running using the run_test.pl script as shown on https://opendds.org/quickstart/GettingStartedWindows.html

The result is as follows:

C:\Users\Test\OpenDDS-3.21\DevGuideExamples\DCPS\Messenger>perl run_test.pl
C:\Users\Test\OpenDDS-3.21\bin\.\DCPSInfoRepo.EXE -ORBDebugLevel 10 -ORBLogFile DCPSInfoRepo.log -o repo.ior
TAO (9080|10592) - Completed initializing the process-wide service context
TAO (9080|10592) - Default ORB services initialization begins
TAO (9080|10592) - ORBInitializer_Registry::register_orb_initializer 0 @012379D8
TAO (9080|10592) - ORBInitializer_Registry::register_orb_initializer 1 @01237434
TAO (9080|10592) - Default ORB services initialization completed
TAO (9080|10592) - We are the default ORB ...
TAO (9080|10592) - Initializing the orb-specific services
.\.\publisher.EXE  -ORBDebugLevel 10 -DCPSDebugLevel 10 -ORBVerboseLogging 1 -DCPSTransportDebugLevel 6  -ORBLogFile publisher.log
.\.\subscriber.EXE  -ORBDebugLevel 10 -DCPSDebugLevel 10 -ORBVerboseLogging 1 -DCPSTransportDebugLevel 6  -ORBLogFile subscriber.log
SampleInfo.sample_rank = 0
SampleInfo.instance_state = ALIVE_INSTANCE_STATE
Message: subject    = Review
         subject_id = 99
         from       = Comic Book Guy
         count      = 0
         text       = Worst. Movie. Ever.
SampleInfo.sample_rank = 0
SampleInfo.instance_state = ALIVE_INSTANCE_STATE
Message: subject    = Review
         subject_id = 100
         from       = Comic Book Guy
         count      = 1
         text       = Worst. Movie. Ever.
SampleInfo.sample_rank = 0
SampleInfo.instance_state = ALIVE_INSTANCE_STATE
Message: subject    = Review
         subject_id = 101
         from       = Comic Book Guy
         count      = 2
         text       = Worst. Movie. Ever.
SampleInfo.sample_rank = 0
SampleInfo.instance_state = ALIVE_INSTANCE_STATE
Message: subject    = Review
         subject_id = 102
         from       = Comic Book Guy
         count      = 3
         text       = Worst. Movie. Ever.
SampleInfo.sample_rank = 0
SampleInfo.instance_state = ALIVE_INSTANCE_STATE
Message: subject    = Review
         subject_id = 103
         from       = Comic Book Guy
         count      = 4
         text       = Worst. Movie. Ever.
SampleInfo.sample_rank = 0
SampleInfo.instance_state = ALIVE_INSTANCE_STATE
Message: subject    = Review
         subject_id = 104
         from       = Comic Book Guy
         count      = 5
         text       = Worst. Movie. Ever.
SampleInfo.sample_rank = 0
SampleInfo.instance_state = ALIVE_INSTANCE_STATE
Message: subject    = Review
         subject_id = 105
         from       = Comic Book Guy
         count      = 6
         text       = Worst. Movie. Ever.
SampleInfo.sample_rank = 0
SampleInfo.instance_state = ALIVE_INSTANCE_STATE
Message: subject    = Review
         subject_id = 106
         from       = Comic Book Guy
         count      = 7
         text       = Worst. Movie. Ever.
SampleInfo.sample_rank = 0
SampleInfo.instance_state = ALIVE_INSTANCE_STATE
Message: subject    = Review
         subject_id = 107
         from       = Comic Book Guy
         count      = 8
         text       = Worst. Movie. Ever.
SampleInfo.sample_rank = 0
SampleInfo.instance_state = ALIVE_INSTANCE_STATE
Message: subject    = Review
         subject_id = 108
         from       = Comic Book Guy
         count      = 9
         text       = Worst. Movie. Ever.
SampleInfo.sample_rank = 0
SampleInfo.instance_state = NOT_ALIVE_DISPOSED_INSTANCE_STATE
SampleInfo.sample_rank = 0
SampleInfo.instance_state = NOT_ALIVE_DISPOSED_INSTANCE_STATE
SampleInfo.sample_rank = 0
SampleInfo.instance_state = NOT_ALIVE_DISPOSED_INSTANCE_STATE
SampleInfo.sample_rank = 0
SampleInfo.instance_state = NOT_ALIVE_DISPOSED_INSTANCE_STATE
SampleInfo.sample_rank = 0
SampleInfo.instance_state = NOT_ALIVE_DISPOSED_INSTANCE_STATE
SampleInfo.sample_rank = 0
SampleInfo.instance_state = NOT_ALIVE_DISPOSED_INSTANCE_STATE
SampleInfo.sample_rank = 0
SampleInfo.instance_state = NOT_ALIVE_DISPOSED_INSTANCE_STATE
SampleInfo.sample_rank = 0
SampleInfo.instance_state = NOT_ALIVE_DISPOSED_INSTANCE_STATE
SampleInfo.sample_rank = 0
SampleInfo.instance_state = NOT_ALIVE_DISPOSED_INSTANCE_STATE
SampleInfo.sample_rank = 0
SampleInfo.instance_state = NOT_ALIVE_DISPOSED_INSTANCE_STATE
INFO: C:\Users\Test\OpenDDS-3.21/bin/DCPSInfoRepo being killed.

C:\Users\Test\OpenDDS-3.21\DevGuideExamples\DCPS\Messenger>
